be/src/storage/publish_version_manager.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <condition_variable>
+#include <map>
+#include <mutex>
+#include <set>
+
+#include "common/status.h"
+#include "gen_cpp/MasterService_types.h"
+#include "util/threadpool.h"
+
+namespace starrocks {
+
+using FinishTaskRequestPtr = std::shared_ptr<TFinishTaskRequest>;
+
+class PublishVersionManager {
+public:
+    Status init();
+    ~PublishVersionManager();
+    Status finish_publish_task(std::vector<TFinishTaskRequest> finish_task_requests);
```

Comment:
From the implementation, the `finish` can not guarantee the task to be finished. It's not a good name for this function.

Reply:
> From the implementation, the `finish` can not guarantee the task to be finished. It's not a good name for this function.

If publish task failed or timeout, we will return the status to FE but not return success and client will get the publish task status. In my opinion, finish means we will return the publish task status to FE, the task status can be success, failed or timeout but not only success.